code/modules/wiremod/components/sensors/view_sensor.dm

```diff
@@ -4,25 +4,28 @@
  * Returns all movable objects in view.
  */
 
-#define VIEW_SENSOR_RANGE 5
-
 /obj/item/circuit_component/view_sensor
 	display_name = "View Sensor"
-	desc = "Outputs a list with all movable objects in it's view. Requires a shell."
+	desc = "Outputs a list with all movable objects in it's view. Requires a shell. Max range of 5 tiles."
 	category = "Sensor"
 
 	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
 
 	power_usage_per_input = 10 //Normal components have 1
 
+	///Allows setting the range of the view sensor
+	var/datum/port/input/range
```

Comment:
Hm isn't the default value of this going to be zero, even before my changes to use clamp? You should make it default to max range

Reply:
Figured out how to do it, and now done